src/useSpeechRecognition.js

```diff
@@ -24,6 +25,14 @@ const useSpeechRecognition = (props = {}) => {
     onResult(transcript);
   };
 
+  const handleError = (event) => {
+    if (event.error === "not-allowed") {
+        //this prevents an infinite loop of error handling:
```

Comment:
The infinite loop is caused by the `onend` handler that is attached in the listen function below. This makes sure the recognition process is started up again automatically after inactivity.

Rather than creating a brand new instance, we should be able to remove the onend handler. I think we'd probably also want to set listening to false as well.

```js
recognition.current.onend = () => {};
setListening(false);
```

Reply:
Much better solution! Fixed.